### PR TITLE
Dismissing ROLE_NONE in RoleDefintion

### PR DESCRIPTION
**Pull request by icreated**
ROLE_NONE is dropped from Role Definition so not possible to use it in AccessControl
As we need it in default param, we check it in setDefault method & verifyDefault

### CODE REVIEW

**Review by vcarluer — APPROVED** [review]
lgtm

File: src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt [hunk 1]
```diff
@@ -190,22 +192,22 @@ open class CsmRbac(
...
+      role: String?,
       rolesDefinition: MutableMap<String, List<String>>
   ): List<String> {
     return rolesDefinition[role] ?: listOf()
   }
 
-  internal fun getUserRole(rbacSecurity: RbacSecurity, user: String): String {
-    return rbacSecurity.accessControlList.firstOrNull { it.id == user }?.role ?: ROLE_NONE
+  internal fun getUserRole(rbacSecurity: RbacSecurity, user: String): String? {
+    return rbacSecurity.accessControlList.firstOrNull { it.id == user }?.role
```
Comment (jreynard-code) [inline src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt:210]: Here if the user is not defined in ACL, the fallback right should be the default one a.k.a : `rbacSecurity.default`

File: src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt [hunk 2]
```diff
@@ -171,8 +173,8 @@ open class CsmRbac(
       rolesDefinition: RolesDefinition
   ): Boolean {
     logger.debug("RBAC ${rbacSecurity.id} - Verifying default roles for permission: $permission")
-    val isAuthorized =
-        this.verifyPermissionFromRole(permission, rbacSecurity.default, rolesDefinition)
+    val defaultRole = if (rbacSecurity.default == ROLE_NONE) null else rbacSecurity.default
```
Comment (jreynard-code) [inline src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt:176]: I don't think that you need this line :thinking: 
If you leave `ROLE_NONE` as `defaultRole` value:
- no need to consider that role is nullable thought all verify* method calls
- this.getRolePermissions("ROLE_NONE", rolesDefinition.permissions) will returned an empty list
- verifyPermission(permission: String, userPermissions: List<String>) will return false if userPermissions is an empty list (which is the expected behavior)

File: src/main/kotlin/com/cosmotech/api/rbac/model/RbacAccessControl.kt [hunk 3]
```diff
@@ -2,6 +2,6 @@
 // Licensed under the MIT license.
 package com.cosmotech.api.rbac.model
 
-import com.cosmotech.api.rbac.ROLE_NONE
+import com.cosmotech.api.rbac.ROLE_VIEWER
 
-open class RbacAccessControl(open var id: String = "emptyId", open var role: String = ROLE_NONE)
+open class RbacAccessControl(open var id: String = "emptyId", open var role: String = ROLE_VIEWER)
```
Comment (jreynard-code) [inline src/main/kotlin/com/cosmotech/api/rbac/model/RbacAccessControl.kt:7]: I think that you can remove the default value.
`role` should be mandatory (like `id`) maybe the domain specific definition (on openapi side) and mappers should be adapt

**Review by jreynard-code — CHANGES_REQUESTED** [review]
Some remarks:

- Nothing forbidden to apply `ROLE_NONE` to user thought `setUserRole` method. This should not be allowed. Could you please add a check if `setUserRole` with `ROLE_NONE` is called? Throw an error if it's the case

File: src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt (hunk 1 above)
Comment (icreated) [reply src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt:233]: Discussed & resolved by call

File: src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt (hunk 2 above)
Comment (icreated) [reply src/main/kotlin/com/cosmotech/api/rbac/CsmRbac.kt:199]: Discussed & resolved by call

**Comment by icreated** [comment]
The role NONE can't pass because missing in Role Definition

**Pull request merged**